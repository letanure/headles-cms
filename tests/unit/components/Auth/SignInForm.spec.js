import { shallowMount } from '@vue/test-utils'
import Component from '@/components/Auth/SignInForm.vue'

describe(`${Component.name}.vue`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Component, {
      propsData: {},
      mocks: {
        $t: (msg) => msg,
      },
    })
  })

  describe('smoke tests', () => {
    it('should render', () => expect(wrapper.exists()).toBe(true))
    it('matches snapshot', () => expect(wrapper.element).toMatchSnapshot())
  })
})
