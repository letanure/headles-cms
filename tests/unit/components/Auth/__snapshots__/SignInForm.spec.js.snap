// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm.vue smoke tests matches snapshot 1`] = `
<el-form-stub
  class="SignInForm"
  labelposition="top"
  labelsuffix=""
  model="[object Object]"
  novalidate="novalidate"
  rules="[object Object]"
  showmessage="true"
  statusicon="true"
  validateonrulechange="true"
>
  <el-form-item-stub
    inlinemessage=""
    label="SignInForm.fields.email.label"
    prop="email"
    showmessage="true"
  >
    <el-input-stub
      autocomplete="off"
      type="email"
      validateevent="true"
      value=""
    />
  </el-form-item-stub>
  <el-form-item-stub
    inlinemessage=""
    label="SignInForm.fields.password.label"
    prop="password"
    showmessage="true"
  >
    <el-input-stub
      autocomplete="off"
      placeholder="SignInForm.fields.password.placeholder"
      showpassword="true"
      type="password"
      validateevent="true"
      value=""
    />
  </el-form-item-stub>
  <el-form-item-stub
    inlinemessage=""
    showmessage="true"
  >
    <el-button-stub
      icon=""
      nativetype="submit"
      type="primary"
    >
      SignInForm.actions.submit.label
    </el-button-stub>
  </el-form-item-stub>
</el-form-stub>
`;
